core/server/data/schema/commands.js

```diff
@@ -145,6 +145,89 @@ async function dropUnique(tableName, columns, transaction) {
     }
 }
 
+/**
+ * Checks if a foreign key exists in a table over the given columns.
+ *
+ * @param {string} fromTableName - name of the table to add the foreign key to
+ * @param {string} fromColumn - column of the table to add the foreign key to
+ * @param {string} toTableName - name of the table to point the foreign key to
+ * @param {string} toColumn - column of the table to point the foreign key to
+ * @param {Object} transaction - connection object containing knex reference
+ * @param {Object} transaction.knex - knex instance
+ */
+async function hasForeign(fromTable, fromColumn, toTable, toColumn, transaction) {
+    const knex = (transaction || db.knex);
+    const client = knex.client.config.client;
+
+    if (client === 'mysql') {
+        const dbName = knex.client.config.connection.database;
+        const [rawConstraints] = await knex.raw(`
+            SELECT i.TABLE_NAME, k.COLUMN_NAME, k.REFERENCED_TABLE_NAME, k.REFERENCED_COLUMN_NAME
+            FROM information_schema.TABLE_CONSTRAINTS i
+            INNER JOIN information_schema.KEY_COLUMN_USAGE k ON i.CONSTRAINT_NAME = k.CONSTRAINT_NAME
+            WHERE i.CONSTRAINT_TYPE = 'FOREIGN KEY'
+            AND i.CONSTRAINT_SCHEMA=:dbName
+            AND i.TABLE_NAME = :fromTable
+            AND k.COLUMN_NAME = :fromColumn
+            AND k.REFERENCED_TABLE_NAME = :toTable
+            AND k.REFERENCED_COLUMN_NAME = :toColumn`, {dbName, fromTable, fromColumn, toTable, toColumn});
+
+        return rawConstraints.length === 1;
+    } else {
+        const foreignKeys = await knex.raw(`PRAGMA foreign_key_list('${fromTable}');`);
+
+        const hasForeignKey = foreignKeys.some(foreignKey => foreignKey.table === toTable && foreignKey.from === fromColumn && foreignKey.to && toColumn);
```

Comment:
what's the purpose of `toColumn` in the end of this condition? Was it meant to be `foreignKey.to === toColumn`? :thinking: